### ISSUE TITLE
nix registry: sort out sensible options

### ISSUE THREAD

**Issue by PkmX** [NixOS/nix#4780]
There are oddities with the accepted/unaccepted options in the `nix registry` subcommand:

* `nix registry list --help` says that it could print a registry entry's type as `flags` if it is specified by `--override-flake`, but it actually doesn't accept such an option:

    ```
    $ nix registry list --override-flake foo bar
    error: unrecognised flag '--override-flake'
    ```

* `nix registry list` should also accept `--json` to produce result in JSON format for other programs to consume.

* `nix registry add/remove/pin` accepts some options like `--arg`/`--argstr`/`--include`/`--impure` that AFAIK doesn't make much sense in this context.

**Comment by akirak** [comment]
I'm interested in this. `nix registry list` doesn't accept `--registry` option, while `add`, `pin`, and `remove` do. How can you display the entries of a registry file in a concise format?

**Comment by Malix-Labs** [comment]
Absolutely, it's extremely subpar as it is currently and needs to be fixed

The fact that you cannot `list` a local registry file using `--registry` or even just filter which registry type to take into account is also a borderline buggy behavior.

And `--override-flake` not working but being documented is definitely a bug.

Some unrelated [Common evaluation options](https://nix.dev/manual/nix/latest/command-ref/new-cli/nix3-registry-add.html#common-evaluation-options) and [Miscellaneous global options](https://nix.dev/manual/nix/latest/command-ref/new-cli/nix3-registry-add.html#miscellaneous-global-options) shouldn't exist too

https://nix.dev/manual/nix/latest/command-ref/new-cli/nix3-registry-list.html

**Comment by Malix-Labs** [comment]
Btw, I created #15660 to generalize the `--registry` option (now called `--file`)

I will also create another PR to make it possible to filter which registry type (https://github.com/NixOS/nix/blob/b3b46091f67cce0ddcc8d99cc1a6ba0233e6a174/src/libfetchers/include/nix/fetchers/registry.hh#L16-L22) to take into account soon
